ddl/index.go

```diff
@@ -131,6 +131,54 @@ func checkIndexPrefixLength(columns []*model.ColumnInfo, idxColumns []*model.Ind
 	return nil
 }
 
+//The function is same as function checkIndexColumn, but the parameter is model.IndexColumn but not ast.IndexPartSpecification.
+func checkIndexInModifiableColumn(col *model.ColumnInfo, ic *model.IndexColumn) error {
```

Comment:
Could you please unify these two functions with 
```go
func checkIndex(col *model.ColumnInfo, prefixLen int)
```
to comply with the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)?

Reply:
good suggestion, change as suggestion.